### PR TITLE
Added voice commands (NO CONFLICT)

### PR DESCRIPTION
**Pull request by Arze1**

### CODE REVIEW

File: commands/voice/play.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        
+        const dispatcher = connnection.playStream(stream);
+        this.stream.set(message.guild.id, dispatcher);
+        
+        dispatcher.on('end', () => {
+          if (this.songQueue.has(message.guild.id) && this.songQueue.get(message.guild.id).length !== 0) {
+            let currentSong = this.songQueue.get(message.guild.id)[0];
+
+            playSong(currentSong.url);
+            message.channel.send(":white_check_mark: SUCCESS\nplaying video `" + currentSong.video.title + "` By `" + currentSong.video.author + "`\n Requested by " + currentSong.user + "\nURL: " + currentSong.url);
```
Comment (ConnorMcF) [inline commands/voice/play.js]: Fix case

File: commands/voice/play.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        }
+      if (this.stream.has(message.guild.id)) {
+        message.channel.send(`:white_check_mark: SUCCESS\nAdded video to queue \`${currentSong.video.title}\` By \`${currentSong.video.author}\`\n Requested by \`${currentSong.user}\`\nURL:  https://www.youtube.com/watch?v=${video.id.videoId}`);
+        
+        let queue = (this.songQueue.get(message.guild.id) || []);
+        queue.push(currentSong);
+        
+        this.songQueue.set(message.guild.id, queue);
+      } else {
+        message.channel.send(`:white_check_mark: SUCCESS\nplaying video \`${currentSong.video.title}\` By \`${currentSong.video.author}\`\n Requested by \`${currentSong.user}\`\nURL:  https://www.youtube.com/watch?v=${video.id.videoId}`);
```
Comment (ConnorMcF) [inline commands/voice/play.js]: Fix case

File: shardingManager.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ if(process.argv.includes('--prepare')) {
 
 	const shardManager = new api.ShardingManager('./shardScript.js', {
 		token: discordCfg.token,
-		totalShards: 2
+		totalShards: 1
```
Comment (ConnorMcF) [inline shardingManager.js]: Add to a config?

File: commands/voice/play.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  args: '(video search)',
+  cooldown: 20000,
+  run: async function(message, args) {
+    const voiceChannel = message.member.voiceChannel;
+
+    const url = "https://www.googleapis.com/youtube/v3/search?type=video&part=snippet&q=" + encodeURI(args.join(",")) + "&key=" + encodeURI(this.botCfg.youtubeApiKey)
+    const ytRequest = await this.request(url, {
+      method: 'GET'
+    });
+    try {
```
Comment (ConnorMcF) [inline commands/voice/play.js]: No need to try catch the whole block, just do the JSON.parse unless anything else likes to throw

File: package.json [hunk 5]
```diff
@@ -25,7 +26,10 @@
     "humanize-duration": "^3.10.0",
     "jimp": "^0.2.28",
     "node-fetch": "^1.7.0",
+    "node-opus": "^0.2.6",
     "pg": "^6.2.3",
-    "speedtest-net": "^1.3.1"
+    "speedtest-net": "^1.3.1",
+    "ytdl": "^0.6.0",
```
Comment (ConnorMcF) [inline package.json]: Is ytdl and ytdl-core needed? Isn't ytdl-core a dependency of ytdl?

Comment (Arze1) [inline package.json]: other way around u need ytdl-core not ytdl

Comment (Arze1) [inline package.json]: ill fix now

File: shardingManager.js (hunk 3 above)
Comment (Arze1) [inline shardingManager.js]: ok

**Comment by ConnorMcF** [comment]
Yeah, what I meant.

File: package.json (hunk 5 above)
Comment (ConnorMcF) [inline package.json]: Yeah, what I meant.

File: commands/voice/play.js (hunk 4 above)
Comment (Arze1) [inline commands/voice/play.js]: well just incase, ive been getting some errors with ytdl so ill keep it

File: commands/voice/play.js (hunk 1 above)
Comment (Arze1) [inline commands/voice/play.js]: dont see any error here ;?

File: commands/voice/play.js (hunk 2 above)
Comment (Arze1) [inline commands/voice/play.js]: whats the fix here?

**Comment by xmatmen** [comment]
I changed some functionality and added some features, see b26f44e4ac1c8d7018cce22f2b5804e80a39e960

**Pull request merged**